compiler/passes/parallel.cpp

```diff
@@ -278,7 +328,11 @@ static Symbol* insertAutoCopyDestroyForTaskArg
           // (But only once per function for each affected argument.)
           Symbol* formal = actual_to_formal(arg);
           CallExpr* autoDestroyCall = new CallExpr(autoDestroyFn,formal);
-          fn->insertBeforeDownEndCount(autoDestroyCall);
+          // Sometimes there is _downEndCount, sometimes there isn't.
+          if (fn->hasFlag(FLAG_ON) && !fn->hasFlag(FLAG_NON_BLOCKING))
+            fn->insertBeforeReturn(autoDestroyCall);
+          else
+            fn->insertBeforeDownEndCount(autoDestroyCall, true);
```

Comment:
Another TODO is to see if the descend[Into]LocalBlocs argument can be replaced with "true", and then remove this extra complexity.